crates/fuzzing/src/oracles/engine.rs

```diff
@@ -93,6 +109,76 @@ pub fn setup_engine_runtimes() {
     crate::oracles::diff_spec::setup_ocaml_runtime();
 }
 
+/// Build a list of allowed values from the given `defaults` using the
+/// `env_list`.
+///
+/// ```
+/// # use wasmtime_fuzzing::oracles::engine::build_allowed_env_list;
+/// // Passing no `env_list` returns the defaults:
+/// assert_eq!(build_allowed_env_list(None, &["a"]), vec!["a"]);
+/// // We can build up a subset of the defaults:
+/// assert_eq!(build_allowed_env_list(Some(vec!["b".to_string()]), &["a","b"]), vec!["b"]);
+/// // Alternately we can subtract from the defaults:
+/// assert_eq!(build_allowed_env_list(Some(vec!["-a".to_string()]), &["a","b"]), vec!["b"]);
+/// ```
+/// ```should_panic
+/// # use wasmtime_fuzzing::oracles::engine::build_allowed_env_list;
+/// // We are not allowed to mix set "addition" and "subtraction"; the following
+/// // will panic:
+/// build_allowed_env_list(Some(vec!["-a".to_string(), "b".to_string()]), &["a", "b"]);
+/// ```
+/// ```should_panic
+/// # use wasmtime_fuzzing::oracles::engine::build_allowed_env_list;
+/// // This will also panic if invalid values are used:
+/// build_allowed_env_list(Some(vec!["c".to_string()]), &["a", "b"]);
+/// ```
+pub fn build_allowed_env_list<'a>(
+    env_list: Option<Vec<String>>,
+    defaults: &[&'a str],
+) -> Vec<&'a str> {
+    if let Some(configured) = &env_list {
+        // Check that the names are either all additions or all subtractions.
+        let subtract_from_defaults = configured.iter().all(|c| c.starts_with("-"));
+        let add_from_defaults = configured.iter().all(|c| !c.starts_with("-"));
+        let keep = if subtract_from_defaults { 1.. } else { 0.. };
```

Comment:
I forgot that `1..` and `..` have different types. So I'd change my suggestion here to:

```rust
let start = if subtract_from_defaults { 1 } else { 0 };
```

and then using `&c[start..]` everywhere, rather than cloning ranges. I think that'll be a little easier to read.